src/org/pentaho/reporting/platform/plugin/cache/FileSystemCacheBackend.java

```diff
@@ -51,64 +54,89 @@ public void setCachePath( final String cachePath ) {
 
   @Override
   public boolean write( final List<String> key, final Serializable value ) {
-    final File file = new File( cachePath + StringUtils.join( cleanKey( key ), File.separator ) );
-
-    final ObjectOutputStream oos;
-    final FileOutputStream fout;
-    try {
-      file.getParentFile().mkdirs();
-      if ( !file.exists() ) {
-        file.createNewFile();
+    final List<String> cleanKey = cleanKey( key );
```

Comment:
Can we rename the "cleanKey" method to "sanitizeKeySegments"? Clean-Key reads as if you remove content (ie "List#clear()"). Sanitize (as in "sanitizing input" ) is a more correct description of what the method does.